orchagent/debug_counter/debug_counter.h

```diff
@@ -0,0 +1,66 @@
+#ifndef SWSS_UTIL_DEBUG_COUNTER_H_
+#define SWSS_UTIL_DEBUG_COUNTER_H_
+
+#include <string>
+#include <unordered_map>
+#include <unordered_set>
+
+extern "C" {
+#include "sai.h"
+}
+
+// Supported debug counter attributes.
+#define COUNTER_ALIAS       "alias"
+#define COUNTER_TYPE        "type"
+#define COUNTER_DESCRIPTION "desc"
+#define COUNTER_GROUP       "group"
+
+// Set of supported attributes to support easy look-up.
+static const std::unordered_set<std::string> supported_debug_counter_attributes =
+{
+    COUNTER_ALIAS,
+    COUNTER_TYPE,
+    COUNTER_DESCRIPTION,
+    COUNTER_GROUP
+};
+
+// Supported debug counter types.
+#define PORT_INGRESS_DROPS   "PORT_INGRESS_DROPS"
+#define PORT_EGRESS_DROPS    "PORT_EGRESS_DROPS"
+#define SWITCH_INGRESS_DROPS "SWITCH_INGRESS_DROPS"
+#define SWITCH_EGRESS_DROPS  "SWITCH_EGRESS_DROPS"
+
+static const std::unordered_map<std::string, sai_debug_counter_type_t> debug_counter_type_lookup = {
+    { PORT_INGRESS_DROPS, SAI_DEBUG_COUNTER_TYPE_PORT_IN_DROP_REASONS },
+    { PORT_EGRESS_DROPS,  SAI_DEBUG_COUNTER_TYPE_PORT_OUT_DROP_REASONS },
+    { SWITCH_INGRESS_DROPS, SAI_DEBUG_COUNTER_TYPE_SWITCH_IN_DROP_REASONS },
+    { SWITCH_EGRESS_DROPS, SAI_DEBUG_COUNTER_TYPE_SWITCH_OUT_DROP_REASONS }
+};
```

Comment:
I only see this static variable used in debug_counter.cpp. I think it is best define this variable in the cpp file.

If in the future this static variable needs to be shared, you can add helper get function to provide access.

Reply:
It's used in debugcounterorch.cpp as well but I like your suggestion better. :)